vitrina/admin.py

```diff
@@ -0,0 +1,136 @@
+import json
+from typing import Any, Dict, Optional
+
+from django.contrib import admin
+from django.contrib.admin.models import LogEntry
+from django.http import HttpRequest
+from django.urls import reverse, NoReverseMatch
+from django.utils.html import format_html, escape
+import reversion
+from reversion.models import Revision, Version
+from reversion.admin import VersionAdmin
+
+from vitrina.utils import RevisionComment, RevisionSource
+
+
+class VersionInline(admin.TabularInline):
+    model = Version
+    extra = 0
+    can_delete = False
+
+    fields = (
+        "version_data",
+        "col_content_type",
+        "col_object_id",
+        "col_object_repr",
+    )
+    readonly_fields = fields
+
+    def has_add_permission(self, request: HttpRequest, obj: Optional[Revision] = None) -> bool:
+        return False
+
+    def has_change_permission(self, request: HttpRequest, obj: Optional[Revision] = None) -> bool:
+        return False
+
+    @admin.display(description="Versijos duomenys")
+    def version_data(self, obj: Version) -> str:
+        data = obj.field_dict
+        pretty = json.dumps(data, indent=2, ensure_ascii=False, default=str)
+
+        element_id = f"snapshot-{obj.pk}"
+
+        return format_html(
+            (
+                '<a href="#" onclick="'
+                "var el=document.getElementById('{}');"
+                "if(el.style.display==='none' || !el.style.display){{"
+                "el.style.display='block';"
+                "this.innerText='Slėpti &#9650;';"
+                "}}else{{"
+                "el.style.display='none';"
+                "this.innerText='Rodyti &#9660;';"
+                "}}"
+                "return false;"
+                '">Rodyti &#9660;</a>'
+                '<div id="{}" style="display:none; margin-top:4px;">'
+                "<pre style='white-space:pre-wrap; margin:0;'>{}</pre>"
+                "</div>"
+            ),
+            element_id,
+            element_id,
+            escape(pretty),
+        )
+
+    @admin.display(description="Modelis")
+    def col_content_type(self, obj: Version) -> Any:
```

Comment:
Let's use full names where possible


```suggestion
    def column_content_type(self, obj: Version) -> Any:
```

Reply:
https://github.com/atviriduomenys/katalogas/pull/2179/commits/447fd4af8020e9143139e7a5ee3e7526561d595e